scripts/globals/abilities/rune_enchantment.lua

```diff
@@ -0,0 +1,56 @@
+-----------------------------------
+-- Ability: Rune Enchantment
+-- Allows you to harbor runes
+-- Obtained: Rune Fencer Level 5
+-- Recast Time: 0:10
+-- Duration: 5:00
+-----------------------------------
+require("scripts/globals/status")
+require("scripts/globals/msg")
+-----------------------------------
+local ability_object = {}
+
+ability_object.onAbilityCheck = function(player, target, ability)
+    return 0, 0
+end
+
+ability_object.onUseAbility = function(user, target, ability)
+    -- Leave blank.
+end
+
+-- helpers for runes
+function getRUNLevel(player)
+    return player:getMainJob() == xi.job.RUN and player:getMainLvl() or player:getSubLvl()
+end
+
+function applyRuneEnhancement(effectType, player)
+    local RUNLevel = getRUNLevel(player)
+    local meritBonus = (player:getMerit(xi.merit.MERIT_RUNE_ENHANCE) * 2) -- 2 more elemental resistance per merit for a maximum total of (2*5) = 10
+    local jobPointBonus = player:getJobPointLevel(xi.jp.RUNE_ENCHANTMENT_EFFECT) -- 1 more elemental resistance per level for a maximum total of 20
+	
+    -- see https://www.bg-wiki.com/ffxi/Category:Rune
+    local power = math.floor((49 * RUNLevel / 99) + 5.5) + meritBonus  + jobPointBonus
+    player:addStatusEffect(effectType, power, 0, 300)
+end
+
+function enforceRuneCounts(target)
+    local RUNLevel = getRUNLevel(target)
+    local maxRunes = RUNLevel >= 65 and 3 or RUNLevel >= 35 and 2 or 1
+    local effects = target:getStatusEffects()
+    local runes = { }
+    local i = 0
+
+    for _, effect in ipairs(effects) do
+        type = effect:getType()
+        if type >= xi.effect.IGNIS and type <= xi.effect.TENEBRAE then
+            runes[i+1] = effect
+            i = i + 1
+        end
+    end
+
+    if i >= maxRunes then -- delete the first rune in the list with the least duration
+        target:delStatusEffect(runes[1]:getType())
+    end
+end
```

Comment:
Content-wise, I think this is a fine first step (like we discussed)

Since this is your first PR, I've enabled the CI to run, and we've gotten the following warnings:
```
Checking scripts/globals/abilities/lux.lua        2 warnings

    scripts/globals/abilities/lux.lua:20:5: (W113) accessing undefined variable enforceRuneCounts
    scripts/globals/abilities/lux.lua:21:5: (W113) accessing undefined variable applyRuneEnhancement

Checking scripts/globals/abilities/rune_enchantment.lua 7 warnings

    scripts/globals/abilities/rune_enchantment.lua:22:10: (W111) setting non-standard global variable getRUNLevel
    scripts/globals/abilities/rune_enchantment.lua:26:10: (W111) setting non-standard global variable applyRuneEnhancement
    scripts/globals/abilities/rune_enchantment.lua:27:22: (W113) accessing undefined variable getRUNLevel
    scripts/globals/abilities/rune_enchantment.lua:30:1: (W611) line contains only whitespace
    scripts/globals/abilities/rune_enchantment.lua:[36](https://github.com/LandSandBoat/server/runs/5314496254?check_suite_focus=true#step:7:36):10: (W111) setting non-standard global variable enforceRuneCounts
    scripts/globals/abilities/rune_enchantment.lua:[37](https://github.com/LandSandBoat/server/runs/5314496254?check_suite_focus=true#step:7:37):22: (W113) accessing undefined variable getRUNLevel
    scripts/globals/abilities/rune_enchantment.lua:[44](https://github.com/LandSandBoat/server/runs/5314496254?check_suite_focus=true#step:7:44):9: (W121) setting read-only global variable type
```

Not a big deal, we just need to move some things around. We forbid free-standing global functions, and we want to try and collect job-specific logic into single areas, away from their specific abilities (if the logic is shared). You can see what I did in my old PR:

https://github.com/LandSandBoat/server/pull/327/files#diff-4f58f0511b54088b461de4f479db0ba797729da0b184dd326f80e6bca6153456R1

```lua
...
require("scripts/globals/job_utils/rune_fencer")
-----------------------------------
local ability_object = {}

ability_object.onAbilityCheck = function(player, target, ability)
    return 0, 0
end

ability_object.onUseAbility = function(player, target, ability)
    return xi.job_utils.rune_fencer.runeEnchantment(player, target, ability, xi.effect.TENEBRAE)
end

...
```

Reply:
That makes sense, will do.

Is there some way for me to run the CI on my fork in the future before I make a PR?

Reply:
The CI is pretty simple, it just does the builds, and for style checks it runs each of the scripts in this folder: 

https://github.com/LandSandBoat/server/tree/base/tools/ci

You can install the requirements and then run them with: `./lua.sh <filename> or `./lua.sh` (for whole repo

Reply:
Perfect. Thank you!